Library/Homebrew/config.rb

```diff
@@ -27,7 +27,11 @@ def mkpath
 HOMEBREW_CACHE_FORMULA = HOMEBREW_CACHE+"Formula"
 
 unless defined? HOMEBREW_BREW_FILE
```

Comment:
I agree. Setting it unconditionally from `config.rb` and removing the corresponding code from `brew.rb` sounds like the right thing to do.